instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -431,6 +844,89 @@ def _instrumented_lambda_handler_call(  # noqa pylint: disable=too-many-branches
     )
 
 
+def determine_flush_timeout() -> int:
+    """
+    Determine the flush timeout for the Lambda function.
+
+    Returns: Flush timeout in milliseconds
+    """
+
+    flush_timeout_env = os.environ.get(
+        OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT, None
+    )
+
+    flush_timeout = 30000
+
+    try:
+        if flush_timeout_env is not None:
+            flush_timeout = int(flush_timeout_env)
+    except ValueError:
+        logger.warning(
+            "Could not convert OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT value %s to int",
+            flush_timeout_env,
+        )
+
+    return flush_timeout
+
+
+def is_aws_context_propagation_disabled(
+    disable_aws_context_propagation_override: Any,
+) -> bool:
+    """
+    Determine if AWS context propagation should be disabled.
+
+    Propagation can be disabled by setting the `OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION` environment variable to:
+    `true`, `1`, or `t`.
+    Or, by passing `True` to the `disable_aws_context_propagation` parameter in the `instrument` method.
+
+    Args:
+        disable_aws_context_propagation_override: Flag indicating whether AWS context propagation should be disabled.
+
+    Returns: True if AWS context propagation should be disabled, False otherwise.
+    """
+
+    if not isinstance(disable_aws_context_propagation_override, bool):
+        logger.warning(
+            "disable_aws_context_propagation must be a boolean, got %s",
+            type(disable_aws_context_propagation_override),
+        )
+        disable_aws_context_propagation_override = False
+
+    disable_from_env = False
+
+    # If the override is set, use it, otherwise check the environment variable.
+    if not disable_aws_context_propagation_override:
+        env_value = (
+            os.getenv(OTEL_LAMBDA_DISABLE_AWS_CONTEXT_PROPAGATION, "False")
+            .strip()
+            .lower()
+        )
+        disable_from_env = env_value in ("true", "1", "t")
+
+    return disable_aws_context_propagation_override or disable_from_env
+
+
+def determine_context_extractor(
+    context_extractor_override: Optional[Callable],
+) -> Callable:
+    """
+    Determine the context extractor to use for the Lambda instrumentation.
+    The context extract should accept an EventWrapper and return an OTel Context.
+
+    Args:
+        context_extractor_override: Optional context extractor to use.
+
+    Returns: A context extractor function.
+    """
+
+    if context_extractor_override and isinstance(
```

Comment:
Suggested Change for Improved Readability and Simplicity
```
if callable(context_extractor_override):
    return context_extractor_override
```

Reply:
Good call, change pushed in 3ed3b1cabab25d3c3c3e5865290c842a8bf9b0f6